cmd/soroban-cli/src/utils/contract-init-template/Cargo.toml.removeextension

```diff
@@ -5,7 +5,7 @@ members = [
 ]
 
 [workspace.dependencies]
-soroban-sdk = "20.3.2"
+soroban-sdk = "21.0.0"
```

Comment:
Should it be updated as part of bump-version.yaml?
Alternatively, should we move template to `soroban-templates`? The versions of soroban-sdk in examples are updated as the part of the `soroban-templates` repo, so wouldn't it be easier to bump all versions at once if it's there? We can also pin commit hash in stellar-cli and update the hash when new protocol version is released (and if necessary template change is required)

Reply:
The bump-version action is shared across repos, so it might be hard to modify it to do this. We could make a CI job that errors if this value is inconsistent with the cli major version, that'd be relatively simple to add.

I'm not sure we should change the overall structure of the examples -> templates since the command advertises that you can use any example.

I'll add the CI check so we don't miss this again.

Reply:
I also noticed that we have git hash pinned for examples in e2e test
https://github.com/stellar/stellar-cli/blob/main/.github/workflows/e2e.yml#L63-L64
While in `init` we use latest commit. Do you think we should update `system-test` to just use `stellar init` instead? In this case we don't need another CI job, because the e2e test (should) catch a regression

Reply:
Yes, we should update that job. I suspect we were using a pre-released version of the examples because they needed a tweaked version early on when setting them up to use init.

Reply:
I've added a test that checks the value stays updated, and removed the pinning of the examples.